pkg/providers/oidc/verifier.go

```diff
@@ -107,3 +120,26 @@ func (v *idTokenVerifier) interfaceSliceToString(slice interface{}) []string {
 	}
 	return strings
 }
+
+func (v *idTokenVerifier) verifyEmail(token *oidc.IDToken) (bool, error) {
+
+	var claims struct {
+		Subject  string `json:"sub"`
+		Email    string `json:"email"`
+		Verified *bool  `json:"email_verified"`
+	}
+
+	if err := token.Claims(&claims); err != nil {
+		return false, fmt.Errorf("failed to parse bearer token claims: %v", err)
```

Comment:
Prefer to use `%w` for errors in `fmt.Errorf`
```suggestion
		return false, fmt.Errorf("failed to parse bearer token claims: %w", err)
```